guppylang/checker/core.py

```diff
@@ -175,7 +175,8 @@ class SubscriptAccess:
     ty: Type
     item_expr: ast.expr
     getitem_call: ast.expr | None = None
-    setitem_call: ast.expr | None = None
+    # Store a temp variable for the RHS of an assignment so it can be assigned a port.
+    setitem_call: tuple[ast.expr, ast.expr] | None = None
```

Comment:
I think it would be better to store the temporary as a `Variable` instead of an `ast.expr`

Reply:
Also we could consider creating a

```python
class SetitemCall(NamedTuple):
    """Docstring ..."""

    #: Expression corresponding to the `__setitem__` call
    call: ast.expr

    #: Variable holding the value that should be written to the subscript.
    #: This variable *must be* assigned before compiling th call! 
    value_var: Variable

    def compile(value: ast.expr, ...):
        """Writes the provided value to the specified location by compiling the `__setitem__` call."""
```